controllers/nstemplateset/nstemplatetier_test.go

```diff
@@ -209,3 +253,408 @@ func assertThatTierTemplateIsSameAs(t *testing.T, expected *toolchainv1alpha1.Ti
+			"selector": map[string]interface{}{
+				"annotations": map[string]interface{}{},
+				"labels": map[string]interface{}{
+					"matchLabels": map[string]interface{}{
+						"toolchain.dev.openshift.com/space": "'johnsmith'",
+					},
+				},
+			},
+		},
+	},
+}
+
+// Test helper functions and data structures for TestProcessGoTemplates
+type templateTestCase struct {
+	name          string
+	templates     []string
+	staticParams  []toolchainv1alpha1.Parameter
+	runtimeParams map[string]string
+	filters       []template.FilterFunc
+	expectedCount int
+	expectedError string
+	validate      func(t *testing.T, objects []runtimeclient.Object)
+}
+
+// Helper to create a TierTemplateRevision for testing
+func createTestTTR(name string, templates []string, params []toolchainv1alpha1.Parameter) *toolchainv1alpha1.TierTemplateRevision {
+	templateObjects := make([]runtime.RawExtension, len(templates))
+	for i, tmpl := range templates {
+		templateObjects[i] = runtime.RawExtension{Raw: []byte(tmpl)}
+	}
+
+	return &toolchainv1alpha1.TierTemplateRevision{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: "toolchain-host-operator",
+		},
+		Spec: toolchainv1alpha1.TierTemplateRevisionSpec{
+			TemplateObjects: templateObjects,
+			Parameters:      params,
+		},
+	}
+}
+
+// Helper to create a tierTemplate with TTR
+func createTestTierTemplate(ttr *toolchainv1alpha1.TierTemplateRevision) *tierTemplate {
+	return &tierTemplate{
+		templateRef: "test-template",
+		tierName:    "basic",
+		typeName:    "dev",
+		ttr:         ttr,
+	}
+}
+
+// Template constants to avoid duplication
+const (
+	configMapTemplate = `{
+		"apiVersion": "v1",
+		"kind": "ConfigMap",
+		"metadata": {
+			"name": "config-{{.SPACE_NAME}}",
+			"namespace": "{{.NAMESPACE}}"
+		},
+		"data": {
+			"config": "{{.CONFIG_VALUE}}"
+		}
+	}`
+	// #nosec G101  // This is a test template, not a real secret and does not have any sensitive data or static values passed
+	secTemplate = `{
+		"apiVersion": "v1",
+		"kind": "Secret",
+		"metadata": {
+			"name": "secret-{{.SPACE_NAME}}",
+			"namespace": "{{.NAMESPACE}}"
+		},
+		"data": {
+			"quota": "{{.QUOTA_LIMIT}}",
+			"runtime-param": "{{.RUNTIME_PARAM}}"
+		}
+	}`
+
+	namespaceTemplate = `{
+		"apiVersion": "v1",
+		"kind": "Namespace",
+		"metadata": {
+			"name": "ns-{{.SPACE_NAME}}"
+		}
+	}`
+
+	invalidTemplate = `{
+		"apiVersion": "v1",
+		"kind": "ConfigMap",
+		"metadata": {
+			"name": "config-{{.SPACE_NAME",
+			"namespace": "default"
+		}
+	}`
+
+	invalidFunctionTemplate = `{
+		"apiVersion": "v1",
+		"kind": "ConfigMap",
+		"metadata": {
+			"name": "config-{{.SPACE_NAME | invalid_function}}",
+			"namespace": "default"
+		}
+	}`
+
+	invalidJSONTemplate = `{
+		"apiVersion": "v1",
+		"kind": "ConfigMap",
+		"metadata": {
+			"name": "config-{{.SPACE_NAME}}",
+			"namespace": "default"
+		},
+		"data": {
+			"invalid": "json"
+		}
+	` // Missing closing brace
+)
+
+func TestProcessGoTemplates(t *testing.T) {
```

Comment:
one more thing - it would be nice to have a test with more than 1 object, I think we only have with 1 object.
Also testing the case where both tiertemplate and ttr is set ( current case after we merge this PR ) . WDYT?

Reply:
minor but - I think a better code structure would be to have the `processGoTemplates` moved to the `TestProcessWithTTR` and call `process` function instead of `processGoTemplates` as that is the entry point for all the tiertemplates. 

Also move the test `convertParametersToMap` to it's own test case as it doesn't rely on anything from the parent test currently. 
WDYT?

Reply:
okay, have updated it now PTAL